src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipe.cs

```diff
@@ -16,10 +16,35 @@ internal struct EventPipeEventInstanceData
     {
         internal IntPtr ProviderID;
         internal uint EventID;
+        internal uint ThreadID;
+        internal Int64 TimeStamp;
+        internal Guid ActivityId;
+        internal Guid ChildActivityId;
         internal IntPtr Payload;
         internal uint PayloadLength;
     }
 
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct EventPipeSessionInfo
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct SystemTime
```

Comment:
PLease don't use SystemTime.  It is inconvinient and unnecessary.   
Note that you might not want to even use QueryPerfCounter stufff either.    DateTime.UtcNow now uses the Window GetSystemTimePreciseAsFileTime function (or Linux equivalent), so you can probably just use DateTime everywhere (FileTime and DateTime are almost the same (they are both 100ns granularity, and only differ in an offset).    This gets rid of a bunch of ugly unit conversions.   Just use DateTime everywhere for timestamps, and use the same PAL call we use for DateTime.UtcNow, and all this gets very simple.